### PR TITLE
fix spelling in docs, code comments, strings & non-exported names

### PR DESCRIPTION
**Pull request by cr7pt0gr4ph7**
Fixed all spelling issues that I could find (using an automated spell checker to find the issues, resolution was always chosen manually). I have explicitly excluded exported names/functions from this PR (even those that are only intended for use within Vike itself) to avoid any possible breakages, those can currently be found in the [`cr7pt0gr4ph7/vike:fix-spelling-public`](https://github.com/cr7pt0gr4ph7/vike/tree/fix-spelling-public) branch and will follow in a second PR.

### CODE REVIEW

File: docs/pages/Head/+Page.mdx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { UiFrameworkExtension, ImplementedBy } from '../../components'
 import { Link, Warning } from '@brillout/docpress'
 
-Kind: cumulative.  
+Kind: cumulative.
```
Comment (brillout) [inline docs/pages/Head/+Page.mdx:4]: This isn't a typo. (It has a functionally in markdown.)

File: docs/pages/eject/+Page.mdx [hunk 2]
```diff
@@ -13,8 +13,8 @@ It moves the code of the extension from `/node_modules/vike-some-extension/` to
 
 ## When to eject?
 
-In genreal, before considering ejection, we recommend reaching out to the maintainer of the Vike extension and to consider making a Pull Request.
+In general, before considering ejection, we recommend reaching out to the maintainer of the Vike extension and to consider making a Pull Request.
 
-Only eject as a last resort &mdash; when it's clear that it's the only option.
+Only eject as a last resort &maas; when it's clear that it's the only option.
```
Comment (brillout) [inline docs/pages/eject/+Page.mdx:18]: Not a typo.

**Review by brillout — CHANGES_REQUESTED** [review]

**Comment by brillout** [comment]
Very nice! Although, to avoid merge conflicts, let's revert the typo fixes in the source code (i.e. inside the `vike/` directory).

**Comment by brillout** [comment]
I went ahead and applied the changes and merged. Thank you! Contribution welcome to https://github.com/vikejs/vike/issues/1943.

**Comment by IanWorley** [comment]
> FYI #1943.
> 
> > using an automated spell checker
> 
> Which one did you use? CC @IanWorley

As of right now I went ahead and decide to use [typos-rs-npm](https://github.com/dalisoft/typos-rs-npm)

File: docs/pages/Head/+Page.mdx (hunk 1 above)
Comment (cr7pt0gr4ph7) [reply docs/pages/Head/+Page.mdx]: That was just my editor automatically trimming whitespace (as I've configured it to do). Good to know!

File: docs/pages/eject/+Page.mdx (hunk 2 above)
Comment (cr7pt0gr4ph7) [reply docs/pages/eject/+Page.mdx]: Yeah, that was done by accident

**Comment by cr7pt0gr4ph7** [comment]
> Very nice! Although, to avoid merge conflicts, let's revert the typo fixes in the source code (i.e. inside the `vike/` directory).

These were the main goal of this PR :sweat_smile: (Cause they were annoying me while debugging something...). Any chance to get at least the spelling fixes for comments & user-facing strings in?

**Comment by cr7pt0gr4ph7** [comment]
All merge conflicts should be easily fixable though, because there are no renames that affect multiple files at once in this PR, only local renames.

(And if it is decided to a spell checker in CI, most/all of these would have to be fixed anyway at some point - so better fix them now IMO).

**Pull request merged**